### ISSUE TITLE
Generated E164 are not compliant

### ISSUE THREAD

**Issue by armetiz** [fzaninotto/Faker#1041]
Hi team.

Here an example of generated e164 phone number : +0569741390735

E164 wiki : https://en.wikipedia.org/wiki/E.164
List of country calling codes : https://en.wikipedia.org/wiki/List_of_country_calling_codes

As you can see on country calling codes; there is no country code leading with 0.

Also; the above generated E164 number could not be validated using the Google Phone Number Validator : https://giggsey.com/libphonenumber/?phonenumber=%2B0569741390735&country=&language=&region=

**Comment by fzaninotto** [comment]
ping @daleattree

**Comment by daleattree** [comment]
Hi there,

Thanks for the info, but the numbers being generated are only in the correct format and not necessary legitimate telephone numbers.

@fzaninotto What is your take on this?

**Comment by fzaninotto** [comment]
I agree, although if it's just a matter of changing a `randomDigit` by a `randomDigitNotNull`, I think it's worth the effort.

**Comment by daleattree** [comment]
The method `e164PhoneNumber` is using `numerify` which uses
`randomDigitNotNull`

On Fri, Oct 7, 2016 at 9:52 AM, Francois Zaninotto <notifications@github.com

**Comment by fzaninotto** [comment]
It's using `numerify` with `#` placeholders, but `numerify` also supports `%` for not null numbers:

```php
    /**
     * Replaces all hash sign ('#') occurrences with a random number
     * Replaces all percentage sign ('%') occurrences with a not null number
     *
     * @param  string $string String that needs to bet parsed
     * @return string
     */
    public static function numerify($string = '###')
```

**Comment by fzaninotto** [comment]
Fixed by #1052
